### ISSUE TITLE
Freeze when "Select Active Targets"

### ISSUE THREAD

**Issue by xwvvvvwx** [noseglid/atom-build#447]
# Description

Atom freezes whenever I press `cmd-alt-t` or select `Select Active Targets` in the command palette.

I am on Atom 1.8.0 (OS X 10.11.5), and `Build` is the only enabled plugin.

I am using the Genesis theme (https://atom.io/themes/genesis-ui).

The Genesis theme works fine with all my other plugins. 
(if it is a problem in the theme, please let me know and I will file a bug in the appropriate project)
# Steps
1. Launch atom
2. Press cmd-alt-t
# Outcome
1. Atom is frozen
# Recording

![:(](https://media.giphy.com/media/xT0GqkLacmPQZMVdIc/giphy.gif)

**Comment by noseglid** [comment]
Do you get a crash report or anything actionable?
I'm not sure what to do with this report. It's not crashing for me, I have the exact same setup.

**Comment by davidterry01** [comment]
@noseglid feeling pretty silly now, am also no longer able to reproduce the issue. 

Closed, sorry for taking up your time :/

**Comment by ndugger** [comment]
I'm actually experiencing this issue right now.

My atom is up to day, the build package is up to date... I'm not sure what's happening.

**Comment by icidasset** [comment]
I have the same issue, no idea where it's coming from and no crash reports (haven't waited longer than 5 minutes though).

**Comment by noseglid** [comment]
When did it start to happen? What have changed? Does a restart solve it? Reinstall?

**Comment by icidasset** [comment]
@noseglid I just installed it, along with the `build-stack` package. Restarting atom and reinstalling the package doesn't fix it. Also tried to disable the `build-stack` package, no changes. Tried to use the dev-tools to find the problem, but that freezes too. Anything else I could try?

**Comment by noseglid** [comment]
I'm guessing somethings ends up in an infinite loop. That's normally what happens when everything freezes. Is (at least) 1 CPU running at 100% when it's frozen?

If so, this could be a way to debug further: https://www.joyent.com/blog/mdb-and-node-js. I have never used it for nodejs, unfortunately but something like that is the only way forward I can think of at least.

**Comment by icidasset** [comment]
Yup.

<img width="527" alt="screen shot 2016-06-21 at 12 41 42 pm" src="https://cloud.githubusercontent.com/assets/296665/16226853/a1aef448-37ad-11e6-8071-a97fb7cf8da7.png">

I will try to look into that mdb thing.

**Comment by noseglid** [comment]
Cool! Let me know if there's anything I can do to help from here.

**Comment by ndugger** [comment]
For me, It's a new computer. I installed atom, installed the atom-build package, and hit [ctrl]+[alt]+[t] in order to open the targets list.

So everything is a fresh install on my side of things. Not sure if I'm missing an unknown dependency somehow. I'll look at it more when I get home.

**Comment by ndugger** [comment]
I've looked through the windows event viewer, watched the resource monitor, and I can't seem to figure out what's causing it to freeze up. It won't even open the target menu; it just sits there and does nothing.

**Comment by simonknittel** [comment]
I'm experiencing something similar.

I have two projects with the same gulpfile.babel.js. In one of the project Atoms freezes when I select "Select Active Targets". The only difference between the two projects is, that in the project, where it freezes the gulpfile.babel.js is located in a subdirectory of the project.

Project 1 (works fine)
root
 - gulpfile.babel.js

Project 2 (freezes)
root
 - subdirectory
 - - gulpfile.babel.js

**Comment by noseglid** [comment]
@simonknittel Could possibly be because it's completely lacking a build file. I'll try and review the code to see if there could be any races or something similar there. Thanks for info.

**Comment by black-snow** [comment]
Same issue here, OS X, atom 1.8 (for more spec [see](https://github.com/AtomBuild/atom-build-npm-apm/issues/21)).

Project's files are within a subdirectory plus it has two `package.json`. Feels like that's the issue (for it didn't hang on other projects).

**Comment by smonetti** [comment]
In my case, it's because I was missing .atom-build.yml, in the project root. When I put that back in there it worked.

**Comment by noseglid** [comment]
@smonetti Thanks for the input. It shouldn't freeze even if there is no `.atom-build*` file, but as a workaround it's good until we have this sorted out.

**Comment by oddman621** [comment]
Same issue here. When I press F7(Ctrl+Alt+T) with Makefile, it freezes. I'm using Linux environment. My cpu is dualcore, when freezes after pressed F7, CPU usage goes about 50%.

**Comment by omeryagmurlu** [comment]
Same here on Linux.

**Comment by Abszissex** [comment]
same issue here.
had two projects folders in my Atom. 
one where the package.json was 
-root
  - package.json
  - some folders...

and one where only two .js files (no package.json or similar) were located. removed the second one from my project folders and it works now. Seems like Atom can't handle two projects when one of them is just a simple folder with some .js or so :/

**Comment by AndiDog** [comment]
I didn't debug but can confirm that I have the same situation as @Abszissex: one Atom "project" with multiple folders added to it. Since I have all of my hundred-something repositories in there, I'd like atom-build to only look in the current Git working directory, or only try to search up in the directory structure to the point where a `.atom-build.yml` is found.

**Comment by gevial** [comment]
Same issue here after upgrading to Atom 1.11 from 1.10.
macOS Sierra
I've tried to remove any .atom-build* files, but no luck.

**Comment by gevial** [comment]
Just moved `.atom-build.json` from my homedir to the project root and voila! Issue gone.

**Comment by noseglid** [comment]
Interesting...
My top idea so far is that it has something with the file watching to do (to detect changes to the build configurations). Not sure in what way though.

@gevial Does the issue return if you put the `.atom-build.json` back in your homedir? What about a `.atom-build.yml` or `.atom-build.cson` ?

**Comment by gevial** [comment]
@noseglid Strange, now I've moved my `.atom-build.yml` from project dir to home and still no freezes. Put my previous JSON config to homedir again and still everything's file. Targets are merged from both configs and no freezes at all.

**Comment by noseglid** [comment]
Crappy elusive bug :)
It's surely something intermittent. I wouldn't be surprised if it "just started happening again".
Any additional information anyone who has this bug can provide is greatly appreciated. I think that's the only way to narrow it down. A sure way to reproduce would make quick work of this I think.

**Comment by sarahec** [comment]
Same error here on Ubuntu Linux -- Atom freezes on Select Targets. Adding atom-build.yml fixes it, removing atom-build.yml brings the problem back.

**Comment by jzi040941** [comment]
when my rust project folder is already open, and I did cargo new --myrustproject to make a src , gitignore and something else.
and press ctrl+shift+P to select build target.
Atom is Freeze.
So I reopen the project and selcect build target, then I can build and run. I can see" Hello World."
I thought maybe build package didn't detact the change and project type (such as rust, C#, etc..)
It cause Freeze.
I hope that these comment useful.

**Comment by ShadowLNC** [comment]
Currently experiencing this issue on Windows 10 Pro 64bit, Atom 1.13.0, `build` 0.67.0 - this may be related to #484.

I believe I have no valid build targets, or at least I haven't added any; I added `.atom-build.yml` to one of my projects but it's an empty file and I see an error when it tries to parse it (so still no valid build targets). However, the gear icon in the bottom right of my editor shows a check mark and "Refereshing targets for<path> (0.1s)" - so it clearly believes there is a target (it's a Django project and needs no building).

Calling `build:select-active-target` (via my remapped F5 key) on my keymap.cson file gives the error "Open file is not part of any open project in Atom". Calling it while in a file that is part of a project will cause the freeze.

Using console logging, I have identified that it calls `awaitSelection()` from `targets-view.js` (call originates from `target-manager.js:199`), but the Promise never resolves (no logging in either the `then` or `catch` clauses appeared in the console). After this occurs, 25-30% of my CPU (so a whole core) is in use by an Atom subprocess. No panel is shown, nor any messages about a lack of targets.
